R/subsetting.R

```diff
@@ -87,8 +109,23 @@ get_queries <- function(aoi, area_name, years, models, parameters, scenarios,
   ntime_hist <- grid_ref$ntime_historical
   ntime_model <- grid_ref$ntime_model
   
+  # Match coordinate systems
+  aoi <- sp::spTransform(aoi, sp::CRS(grid_ref$crs))
+  
   # Get relative index positions to full grid
-  index_pos <- get_aoi_indexes(aoi, grid_ref)
+  # slight buffering of extent allows us to handle points
+  if (class(aoi) == "SpatialPointsDataFrame") {
+    orig_crs <- raster::projection(aoi)
+    # buffering is only possible in projected coordinate systems
+    proj_aoi <- sp::spTransform(aoi, sp::CRS("+init=epsg:5070"))
+    extended_aoi <- rgeos::gBuffer(proj_aoi, width=.1)
+    extended_aoi <- sp::spTransform(extended_aoi, orig_crs)
```

Comment:
```suggestion
    aoi <- sp::spTransform(extended_aoi, orig_crs)
```

It would be a little cleaner to just have one `if` statement that redefines the `aoi`, rather than an if-else statement, because the `else` part doesn't do anything.

Reply:
I'll include in next pull request. other issues were identified.